internal/corazawaf/body_buffer.go

```diff
@@ -51,7 +51,7 @@ func (br *BodyBuffer) Write(data []byte) (n int, err error) {
 		return 0, nil
 	}
 
-	if br.length == br.options.Limit || br.length >= (br.options.Limit-int64(len(data))) {
+	if br.length > br.options.Limit || br.length > (br.options.Limit-int64(len(data))) {
```

Comment:
Will `br.length` ever be bigger than the limit? I think the entire logic is that it will never go beyond that.

Reply:
I included the upper limit because there was a failing test (TestTxPhase4Magic).
It has `ResponseBodyLimit = 3` so it expects that `3` bytes are written.
With the latest commit, `3` bytes are now written and the limit is considered reached (written because of process partial).

This is the check that should never be triggered if the Write function has been correctly called following the entire logic

Reply:
Addressing here: https://github.com/corazawaf/coraza/pull/593!